tests/components/bsblan/test_climate.py

```diff
@@ -0,0 +1,299 @@
+    entity_registry: er.EntityRegistry,
+    static_file: str,
+) -> None:
+    """Test the initial parameters in Celsius."""
+    await mock_bsblan.set_static_values(static_file)
+    await setup_with_selected_platforms(hass, mock_config_entry, [Platform.CLIMATE])
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+    climate_entity = hass.data[CLIMATE_DOMAIN].get_entity(ENTITY_ID)
+
+    # Test when current_temperature is "---"
+    with patch.object(
+        mock_bsblan.state.return_value.current_temperature, "value", "---"
+    ):
+        await hass.helpers.entity_component.async_update_entity(ENTITY_ID)
+        state = hass.states.get(ENTITY_ID)
+        assert state.attributes["current_temperature"] is None
+
+    # Test target_temperature
+    target_temperature = 16.2
+    with patch.object(
+        mock_bsblan.state.return_value.target_temperature, "value", target_temperature
+    ):
+        await hass.helpers.entity_component.async_update_entity(ENTITY_ID)
+        state = hass.states.get(ENTITY_ID)
+        assert state.attributes["temperature"] == target_temperature
+
+    # Test hvac_mode when preset_mode is ECO
+    with patch.object(mock_bsblan.state.return_value.hvac_mode, "value", PRESET_ECO):
+        assert climate_entity.hvac_mode == HVACMode.AUTO
+
+    # Test hvac_mode with other values
+    with patch.object(mock_bsblan.state.return_value.hvac_mode, "value", HVACMode.HEAT):
+        assert climate_entity.hvac_mode == HVACMode.HEAT
+
+    # Test preset_mode
+    with patch.object(
+        BSBLANClimate, "hvac_mode", new_callable=PropertyMock
+    ) as mock_hvac_mode:
+        mock_hvac_mode.return_value = HVACMode.AUTO
+        mock_bsblan.state.return_value.hvac_mode.value = PRESET_ECO
+        await hass.helpers.entity_component.async_update_entity(ENTITY_ID)
+        state = hass.states.get(ENTITY_ID)
+        assert state.attributes["preset_mode"] == PRESET_ECO
+
+
+@pytest.mark.parametrize("static_file", ["static.json"])
+@pytest.mark.parametrize(
+    ("mode", "expected_call"),
+    [
+        (HVACMode.HEAT, HVACMode.HEAT),
+        (HVACMode.COOL, HVACMode.COOL),
+        (HVACMode.AUTO, HVACMode.AUTO),
+        (HVACMode.OFF, HVACMode.OFF),
+    ],
+)
+async def test_async_set_hvac_mode(
+    hass: HomeAssistant,
+    mock_bsblan: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+    static_file: str,
+    mode: str,
+    expected_call: str,
+) -> None:
+    """Test the async_set_hvac_mode function."""
+    await mock_bsblan.set_static_values(static_file)
+    await setup_with_selected_platforms(hass, mock_config_entry, [Platform.CLIMATE])
+
+    climate_entity = hass.data[CLIMATE_DOMAIN].get_entity(ENTITY_ID)
+
+    with patch.object(climate_entity, "async_set_data") as mock_set_data:
+        await climate_entity.async_set_hvac_mode(mode)
+        mock_set_data.assert_called_once_with(hvac_mode=expected_call)
```

Comment:
We should use service calls to change the hvac mode and then you can use your `mock_bsblan` to check if it did the right thing